Beam/app/Http/Controllers/ArticleController.php

```diff
@@ -32,11 +33,15 @@ class ArticleController extends Controller
 
     private $conversionRateConfig;
 
-    public function __construct(JournalContract $journal, ConversionRateConfig $conversionRateConfig)
+    private $tokenProvider;
+
+    public function __construct(JournalContract $journal, ConversionRateConfig $conversionRateConfig, TokenProvider $tokenProvider =  null)
```

Comment:
Is there a reason why the `$tokenProvider` is made optional? The initialization in `app/Providers/RempJournalServiceProvider.php` is done in a way that there always should be an implementation.

If not, I'd make it required and remove `if ($this->tokenProvider) {` conditions below.

Reply:
fixed

Reply:
Thanks. Will you please also squash and force push the commits? We'll be merging this in our internal repository which gets mirrored here and if we squash it there, Github won't be able to match your commits with our squash.

When it's squashed, we'll merge this to master.

Reply:
OK, I got it.